File: part1-exercises-for-rust/hardways/leetcode-set-matrix-zeroes/solution.c
void setZeroes(int** matrix, int matrixSize, int* matrixColSize){
    int i, j, k;
    int rRow = -1;
    for (i = 0; i < matrixSize; i++) {
        for (j = 0; j < matrixColSize[i]; j++) {
            if (matrix[i][j] == 0) {
                for (k = 0; k < matrixColSize[i]; k++) {
                    if (matrix[i][k] == 0) {
                        matrix[i][k] = 1;
                    } else {
                        matrix[i][k] = 0;
                    }
                }
                if (rRow != -1) {
                    for (k = 0; k < matrixColSize[rRow]; k++) {
                        matrix[i][k] = matrix[rRow][k] | matrix[i][k];
                        matrix[rRow][k] = 0;
                    }
                }
                rRow = i;
                break;
            }
        }
    }
    if (rRow != -1) {
        for (j = 0; j < matrixColSize[rRow]; j++) {
            if (matrix[rRow][j] == 1) {
                for (k = 0; k < matrixSize; k++) {
                    matrix[k][j] = 0;
                }
            }
        }
    }
}

File: part1-exercises-for-rust/hardways/leetcode-set-matrix-zeroes/solution.rs
impl Solution {
    pub fn set_zeroes(matrix: &mut Vec<Vec<i32>>) {
        let mut a = Vec::<(usize,usize)>::new();
        for i in 0..matrix.len(){
            for j in 0..matrix[0].len(){
                if matrix[i][j] == 0{
                    a.push((i,j));
                }
            }
        }
        for m in a{
            for k in 0..matrix.len(){
                matrix[k][m.1] = 0;
            }
            for k in 0..matrix[0].len(){
                matrix[m.0][k] = 0;
            }
        }
    }
}

// 事实上这个的时间复杂度挺高的，但由于0比较稀疏，可能在实际使用中表现较好